packages/mdc-textfield/label/mdc-text-field-label.scss

```diff
@@ -0,0 +1,91 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "../variables";
+@import "@material/rtl/mixins";
+@import "@material/theme/variables";
+@import "@material/theme/mixins";
+
+@include mdc-text-field-invalid-label-shake_keyframes_(standard, 100%);
+@include mdc-text-field-invalid-label-shake_keyframes_(box, 50%);
+
+// postcss-bem-linter: define text-field-label
+.mdc-text-field {
+  &__label {
+    position: absolute;
+    bottom: 8px;
+    left: 0;
+    transform-origin: left top;
+    transition: mdc-text-field-transition(transform), mdc-text-field-transition(color);
+    color: $mdc-text-field-underline-on-light-idle;
+    cursor: text;
+
+    // stylelint-disable plugin/selector-bem-pattern
+    @include mdc-rtl(".mdc-text-field") {
+      right: 0;
+      left: auto;
+      transform-origin: right top;
+    }
+    // stylelint-enable plugin/selector-bem-pattern
+
+    @include mdc-theme-dark(".mdc-text-field") {
+      @include mdc-theme-prop(color, $mdc-text-field-dark-label);
+    }
+
+    &--float-above {
+      transform: translateY(-100%) scale(.75, .75);
+      cursor: auto;
+    }
+  }
+}
+
+// stylelint-disable plugin/selector-bem-pattern
+
+.mdc-text-field__label--float-above {
+  &.mdc-text-field__label--shake {
+    animation: invalid-shake-float-above-standard 250ms 1;
+  }
+}
+
+// Move label when text-field gets autofilled in Chrome
+.mdc-text-field__input {
+  &:-webkit-autofill + .mdc-text-field__label {
+    transform: translateY(-100%) scale(.75, .75);
+    cursor: auto;
+  }
+}
+
+// stylelint-enable plugin/selector-bem-pattern
+
+.mdc-text-field__input:required + .mdc-text-field__label::after {
+  margin-left: 1px;
+  content: "*";
+
+  .mdc-text-field--focused & {
+    color: $mdc-text-field-error-on-light;
+
+    @include mdc-theme-dark(".mdc-text-field", true) {
+      color: $mdc-text-field-error-on-dark;
+    }
+  }
+}
+
+// mdc-form-field tweaks to align text field label correctly
```

Comment:
I'd leave these in the top-level scss as well since this concerns a separate label element for CSS-only text fields, which isn't governed by this subcomponent.